src/main/java/com/faforever/api/data/domain/User.java

```diff
@@ -1,20 +1,39 @@
 package com.faforever.api.data.domain;
 
+import com.yahoo.elide.annotation.UpdatePermission;
 import lombok.Setter;
 
 import javax.persistence.Column;
 import javax.persistence.Entity;
+import javax.persistence.OneToMany;
 import javax.persistence.Table;
+import java.util.List;
 
 @Entity
 @Table(name = "login")
 @Setter
 public class User extends Login {
 
   private String password;
-  
+  private List<RoleUserAssignment> roleAssignments;
+
   @Column(name = "password")
   public String getPassword() {
     return password;
   }
+
+  public boolean hasPermission(String permission) {
+    return roleAssignments.stream()
+        .anyMatch(roleUserAssignment ->
+            roleUserAssignment.getRole().getPermissionAssignments().stream()
+                .anyMatch(rolePermissionAssignment ->
+                    rolePermissionAssignment.getPermission().getName().equals(permission)));
```

Comment:
I think this can be simplified to:
```
    return roleAssignments.stream()
        .flatMap(roleUserAssignment ->roleUserAssignment.getRole().getPermissionAssignments().stream())
        .anyMatch(rolePermissionAssignment -> permission.equals(rolePermissionAssignment.getPermission().getName()));
```
(And I flipped the `equals()`, see http://stackoverflow.com/questions/43453831/is-calling-equals-on-the-same-object-faster)